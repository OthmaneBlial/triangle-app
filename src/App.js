/**
 * @author Blial Othmane
 *
 * Finding out if a Triangle is is Equilateral, Isosceles or Scalene
 * An equilateral triangle is a triangle where its three sides are equal.
 * An isosceles triangle is a triangle that has two sides of equal length.
 * A scalene triangle is a triangle that all its sides are not equal.
 *
 * Solving the problem while provinding a simple UI:
 * We are asking the user to enter the three sides of the triangle,
 * then, we should check that the user enters a number, and also not let the input empty,
 * otherwise, the submit button should be disabled, so the user will not send wrong data.
 * After that, we should check, based on the three sides values, the type of the triangle,
 * then, we render the result below the form, and reset the form,
 * so the user can enter new values and check out another triangle's type.
 * @param {Object of numbers} values: {sideOne, sideTwo, sideThree} - Triangle's sides
 * @param {string} type - Triangle's type
 */

import React, { Component } from "react";
import { Formik } from "formik";
import * as Yup from "yup";
import "./index.css";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      values: {},
      type: "Equilateral"
    };
  }

  findsOutTriangleType() {
    const { sideOne, sideTwo, sideThree } = this.state.values;
    if (sideOne === sideTwo || sideTwo !== sideThree || sideTwo !== sideThree)
      this.setState({ type: "Isosceles" });
    if (sideOne === sideTwo && sideTwo === sideThree)
      this.setState({ type: "Equilateral" });
    if (sideOne !== sideTwo && sideTwo !== sideThree)
      this.setState({ type: "Scalene" });
  }

  render() {
    return (
      <div className="container" data-ts="Main">
        <div className="form-container">
          <h1 className="title">The Triangle Type</h1>
          <Formik
            initialValues={{ sideOne: 0, sideTwo: 0, sideThree: 0 }}
            validationSchema={Yup.object().shape({
              sideOne: Yup.number().required("Required"),
              sideTwo: Yup.number().required("Required"),
              sideThree: Yup.number().required("Required")
            })}
            onSubmit={(values, { setSubmitting, setErrors, resetForm }) => {
              this.setState(
                {
                  values: values
                },
                () => {
                  this.findsOutTriangleType();
                  setSubmitting(false);
                  resetForm();
                }
              );
            }}
          >
            {props => {
              const {
                values,
                touched,
                errors,
                isSubmitting,
                handleChange,
                handleBlur,
                handleSubmit
              } = props;
              return (
                <form data-ts="Form" onSubmit={handleSubmit}>
                  <h4 className="form-title">
                    Please enter the sides of the Triangle
                  </h4>
                  <fieldset>
                    <label
                      className={
                        errors.sideOne && touched.sideOne ? "ts-error" : ""
                      }
                    >
                      <span>Side One</span>
                      <input
                        type="number"
                        name="sideOne"
                        value={values.sideOne}
                        onChange={handleChange}
                        onBlur={handleBlur}
                      />
                    </label>
                    {errors.sideOne &&
                      touched.sideOne && (
                        <dl className="ts-errors">
                          <dt>You must enter a number</dt>
                        </dl>
                      )}
                  </fieldset>

                  <fieldset>
                    <label
                      className={
                        errors.sideTwo && touched.sideTwo ? "ts-error" : ""
                      }
                    >
                      <span>Side One</span>
                      <input
                        type="number"
                        name="sideTwo"
                        value={values.sideTwo}
                        onChange={handleChange}
                        onBlur={handleBlur}
                      />
                    </label>
                    {errors.sideTwo &&
                      touched.sideTwo && (
                        <dl className="ts-errors">
                          <dt>You must enter a number</dt>
                        </dl>
                      )}
                  </fieldset>

                  <fieldset>
                    <label
                      className={
                        errors.sideThree && touched.sideThree ? "ts-error" : ""
                      }
                    >
                      <span>Side One</span>
                      <input
                        type="number"
                        name="sideThree"
                        value={values.sideThree}
                        onChange={handleChange}
                        onBlur={handleBlur}
                      />
                    </label>
                    {errors.sideThree &&
                      touched.sideThree && (
                        <dl className="ts-errors">
                          <dt>You must enter a number</dt>
                        </dl>
                      )}
                  </fieldset>

                  <button
                    type="submit"
                    data-ts="Button"
                    disabled={
                      isSubmitting ||
                      (errors.sideOne && touched.sideOne) ||
                      (errors.sideTwo && touched.sideTwo) ||
                      (errors.sideThree && touched.sideThree)
                    }
                    className="ts-primary btn-large has-shadow"
                  >
                    <span>Finds out the Triangle Type</span>
                  </button>
                </form>
              );
            }}
          </Formik>
          <div className="triangle-type">
            <h4 className="triangle-type-text">
              This triangle is {this.state.type}
            </h4>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
