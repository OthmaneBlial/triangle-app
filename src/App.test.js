import React from "react";
import ReactDOM from "react-dom";
import App from "./App";

import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("the method should return Equilateral if all sides are equal", () => {
  const wrapper = shallow(<App />);
  const instance = wrapper.instance();
  const values = { values: { sideOne: 20, sideTwo: 20, sideThree: 20 } };
  wrapper.setState(values);
  instance.findsOutTriangleType();
  expect(wrapper.state("type")).toBe("Equilateral");
});

it("the method should return Isosceles if two sides are equal", () => {
  const wrapper = shallow(<App />);
  const instance = wrapper.instance();
  const values = { values: { sideOne: 20, sideTwo: 20, sideThree: 30 } };
  wrapper.setState(values);
  instance.findsOutTriangleType();
  expect(wrapper.state("type")).toBe("Isosceles");
});

it("the method should return Scalene if two sides are equal", () => {
  const wrapper = shallow(<App />);
  const instance = wrapper.instance();
  const values = { values: { sideOne: 10, sideTwo: 20, sideThree: 30 } };
  wrapper.setState(values);
  instance.findsOutTriangleType();
  expect(wrapper.state("type")).toBe("Scalene");
});

it("render the App snapshot", () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});
