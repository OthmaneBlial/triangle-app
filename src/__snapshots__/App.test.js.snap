// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the App snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="form-container"
      >
        <h1
          className="title"
        >
          The Triangle Type
        </h1>
        <Formik
          enableReinitialize={false}
          initialValues={
            Object {
              "sideOne": 0,
              "sideThree": 0,
              "sideTwo": 0,
            }
          }
          isInitialValid={false}
          onSubmit={[Function]}
          validateOnBlur={true}
          validateOnChange={true}
          validationSchema={
            ObjectSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_defaultDefault": [Function],
              "_deps": Array [],
              "_excludedEdges": Array [],
              "_exclusive": Object {},
              "_mutate": false,
              "_nodes": Array [
                "sideThree",
                "sideTwo",
                "sideOne",
              ],
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "object",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "fields": Object {
                "sideOne": NumberSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "number",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "sideThree": NumberSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "number",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "sideTwo": NumberSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "number",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
              },
              "tests": Array [],
              "transforms": Array [
                [Function],
              ],
            }
          }
        >
          [Function]
        </Formik>
        <div
          className="triangle-type"
        >
          <h4
            className="triangle-type-text"
          >
            This triangle is 
            Equilateral
          </h4>
        </div>
      </div>,
      "className": "container",
      "data-ts": "Main",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          >
            The Triangle Type
          </h1>,
          <Formik
            enableReinitialize={false}
            initialValues={
              Object {
                "sideOne": 0,
                "sideThree": 0,
                "sideTwo": 0,
              }
            }
            isInitialValid={false}
            onSubmit={[Function]}
            validateOnBlur={true}
            validateOnChange={true}
            validationSchema={
              ObjectSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_defaultDefault": [Function],
                "_deps": Array [],
                "_excludedEdges": Array [],
                "_exclusive": Object {},
                "_mutate": false,
                "_nodes": Array [
                  "sideThree",
                  "sideTwo",
                  "sideOne",
                ],
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "object",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "fields": Object {
                  "sideOne": NumberSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "number",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                  "sideThree": NumberSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "number",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                  "sideTwo": NumberSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "number",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                },
                "tests": Array [],
                "transforms": Array [
                  [Function],
                ],
              }
            }
          >
            [Function]
          </Formik>,
          <div
            className="triangle-type"
          >
            <h4
              className="triangle-type-text"
            >
              This triangle is 
              Equilateral
            </h4>
          </div>,
        ],
        "className": "form-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Triangle Type",
            "className": "title",
          },
          "ref": null,
          "rendered": "The Triangle Type",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": [Function],
            "enableReinitialize": false,
            "initialValues": Object {
              "sideOne": 0,
              "sideThree": 0,
              "sideTwo": 0,
            },
            "isInitialValid": false,
            "onSubmit": [Function],
            "validateOnBlur": true,
            "validateOnChange": true,
            "validationSchema": ObjectSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_defaultDefault": [Function],
              "_deps": Array [],
              "_excludedEdges": Array [],
              "_exclusive": Object {},
              "_mutate": false,
              "_nodes": Array [
                "sideThree",
                "sideTwo",
                "sideOne",
              ],
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "object",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "fields": Object {
                "sideOne": NumberSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "number",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "sideThree": NumberSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "number",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "sideTwo": NumberSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "number",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
              },
              "tests": Array [],
              "transforms": Array [
                [Function],
              ],
            },
          },
          "ref": null,
          "rendered": [Function],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4
              className="triangle-type-text"
            >
              This triangle is 
              Equilateral
            </h4>,
            "className": "triangle-type",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "This triangle is ",
                "Equilateral",
              ],
              "className": "triangle-type-text",
            },
            "ref": null,
            "rendered": Array [
              "This triangle is ",
              "Equilateral",
            ],
            "type": "h4",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="form-container"
        >
          <h1
            className="title"
          >
            The Triangle Type
          </h1>
          <Formik
            enableReinitialize={false}
            initialValues={
              Object {
                "sideOne": 0,
                "sideThree": 0,
                "sideTwo": 0,
              }
            }
            isInitialValid={false}
            onSubmit={[Function]}
            validateOnBlur={true}
            validateOnChange={true}
            validationSchema={
              ObjectSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_defaultDefault": [Function],
                "_deps": Array [],
                "_excludedEdges": Array [],
                "_exclusive": Object {},
                "_mutate": false,
                "_nodes": Array [
                  "sideThree",
                  "sideTwo",
                  "sideOne",
                ],
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "object",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "fields": Object {
                  "sideOne": NumberSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "number",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                  "sideThree": NumberSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "number",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                  "sideTwo": NumberSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "number",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                },
                "tests": Array [],
                "transforms": Array [
                  [Function],
                ],
              }
            }
          >
            [Function]
          </Formik>
          <div
            className="triangle-type"
          >
            <h4
              className="triangle-type-text"
            >
              This triangle is 
              Equilateral
            </h4>
          </div>
        </div>,
        "className": "container",
        "data-ts": "Main",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="title"
            >
              The Triangle Type
            </h1>,
            <Formik
              enableReinitialize={false}
              initialValues={
                Object {
                  "sideOne": 0,
                  "sideThree": 0,
                  "sideTwo": 0,
                }
              }
              isInitialValid={false}
              onSubmit={[Function]}
              validateOnBlur={true}
              validateOnChange={true}
              validationSchema={
                ObjectSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_defaultDefault": [Function],
                  "_deps": Array [],
                  "_excludedEdges": Array [],
                  "_exclusive": Object {},
                  "_mutate": false,
                  "_nodes": Array [
                    "sideThree",
                    "sideTwo",
                    "sideOne",
                  ],
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "object",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "fields": Object {
                    "sideOne": NumberSchema {
                      "_blacklist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "required": false,
                      },
                      "_mutate": false,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "number",
                      "_typeError": [Function],
                      "_whitelist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "tests": Array [
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                    "sideThree": NumberSchema {
                      "_blacklist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "required": false,
                      },
                      "_mutate": false,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "number",
                      "_typeError": [Function],
                      "_whitelist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "tests": Array [
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                    "sideTwo": NumberSchema {
                      "_blacklist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "required": false,
                      },
                      "_mutate": false,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "number",
                      "_typeError": [Function],
                      "_whitelist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "tests": Array [
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                  },
                  "tests": Array [],
                  "transforms": Array [
                    [Function],
                  ],
                }
              }
            >
              [Function]
            </Formik>,
            <div
              className="triangle-type"
            >
              <h4
                className="triangle-type-text"
              >
                This triangle is 
                Equilateral
              </h4>
            </div>,
          ],
          "className": "form-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "The Triangle Type",
              "className": "title",
            },
            "ref": null,
            "rendered": "The Triangle Type",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": [Function],
              "enableReinitialize": false,
              "initialValues": Object {
                "sideOne": 0,
                "sideThree": 0,
                "sideTwo": 0,
              },
              "isInitialValid": false,
              "onSubmit": [Function],
              "validateOnBlur": true,
              "validateOnChange": true,
              "validationSchema": ObjectSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_defaultDefault": [Function],
                "_deps": Array [],
                "_excludedEdges": Array [],
                "_exclusive": Object {},
                "_mutate": false,
                "_nodes": Array [
                  "sideThree",
                  "sideTwo",
                  "sideOne",
                ],
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "object",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "fields": Object {
                  "sideOne": NumberSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "number",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                  "sideThree": NumberSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "number",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                  "sideTwo": NumberSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "number",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                },
                "tests": Array [],
                "transforms": Array [
                  [Function],
                ],
              },
            },
            "ref": null,
            "rendered": [Function],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h4
                className="triangle-type-text"
              >
                This triangle is 
                Equilateral
              </h4>,
              "className": "triangle-type",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "This triangle is ",
                  "Equilateral",
                ],
                "className": "triangle-type-text",
              },
              "ref": null,
              "rendered": Array [
                "This triangle is ",
                "Equilateral",
              ],
              "type": "h4",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
